app/Http/Controllers/TodoController.php

```diff
@@ -17,15 +17,35 @@ public function index(Request $request): JsonResponse
     {
         $sortBy = $request->query('sort_by', 'title');
         $sortOrder = $request->query('sort_order', 'asc');
+        $searchQuery = $request->query('search_query', '');
 
         $validSortColumns = ['title', 'description', 'date', 'priority'];
         if (!in_array($sortBy, $validSortColumns) || !in_array($sortOrder, ['asc', 'desc'])) {
             return response()->json(['error' => 'Invalid sort parameters'], 400);
         }
 
-        $todos = Todo::query()
-            ->orderBy($sortBy, $sortOrder)
-            ->get();
+        $query = Todo::query();
+
+        if ($sortBy === 'priority') {
```

Comment:
Please hide the filtering logic behind a class. That is a better approach than putting the business logic directly into the controller action.

PS: You can even use packages. They do the "dirty work" for you. Please do some research and try to find a package for sorting and filtering functionality.